tests/python/test_linalg.py

```diff
@@ -446,3 +445,21 @@ def func():
     assert v[1, 0, 0].z == -3
     assert v[1, 0, 0].w == 4
     assert np.allclose(v.to_numpy()[1, 0, 0, :], np.array([6, 0, -3, 4]))
+
+
+@ti.host_arch_only
+def test_diag():
+    m1 = ti.Matrix(3, 3, dt=ti.f32, shape=())
+
+    @ti.kernel
+    def fill():
+        m1[None] = ti.Matrix.diag(dim=3, val=1.4)
+
+    fill()
+
+    for i in range(3):
+        for j in range(3):
+            if i == j:
+                assert m1[None][i, j] == approx(1.4)
```

Comment:
Before this PR `m1[None][i, j] = 1` ...

Reply:
Why? I thought the original code was just OK?